### PR TITLE
Formatting layout

### PR DESCRIPTION
**Pull request by azhang7740**

### CODE REVIEW

File: Flixter/DetailsViewController.m [hunk 1]
```diff
@@ -54,14 +55,13 @@ - (void)viewDidLoad {
...
 }
 
-/*
 #pragma mark - Navigation
 
 // In a storyboard-based application, you will often want to do a little preparation before navigation
 - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
-    // Get the new view controller using [segue destinationViewController].
-    // Pass the selected object to the new view controller.
+    NSString *movieIdString = [NSString stringWithFormat:@"%@", self.movieDetails[@"id"]];
```
Comment (joesus) [inline Flixter/DetailsViewController.m:62]: This has term "Hungarian Notation" and is almost always a bad idea when naming things.

The reason is that you want this variable to be the movie id. If the type changes to an int or a more qualified type of string, you may forget to rename this and now the code is confusing to read.

File: Flixter/MovieViewController.m [hunk 2]
```diff
@@ -38,6 +38,7 @@ - (void)viewDidLoad {
     [self.tableView insertSubview:refreshControl atIndex:0];
 
     [self networkRequest];
+    NSLog(@"%@", self.movieData);
```
Comment (joesus) [inline Flixter/MovieViewController.m:41]: I should explain why this drives me nuts. 
If you do this in an app you're working on with multiple people. At some point you will want to use the logs to debug something. You will add a statement but it will be swallowed into the sea of log statements. So you'll add something like `NSLog(@"ANGELINA ***: %@", self.movieData);` and then filter the logs. But then you'll miss something important like a log statement from a system library that's being deprecated, or a broken constraint that will make things look bad on some device sizes.

tldr; I promise that leaving log statements in is a bad habit that will come back to bite you at some point.

File: Flixter/TrailerViewController.h [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+//
+
+#import <UIKit/UIKit.h>
+#import <WebKit/WebKit.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface TrailerViewController : UIViewController
+
+@property (weak, nonatomic) IBOutlet WKWebView *trailerWebView;
```
Comment (joesus) [inline Flixter/TrailerViewController.h:15]: For some reason it is a convention in UIKit to use hungarian notation for IBOutlets. This is the only time I really see this as an acceptable use case.

File: Flixter/TrailerViewController.m [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+@property (nonatomic, strong) NSArray *movieTrailerData;
+
+@end
+
+@implementation TrailerViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    // Do any additional setup after loading the view.
```
Comment (joesus) [inline Flixter/TrailerViewController.m:21]: nit: boilerplate code comments

File: Flixter/TrailerViewController.m [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    [self trailerNetworkRequest];
+}
+
+- (void)trailerNetworkRequest {
```
Comment (joesus) [inline Flixter/TrailerViewController.m:25]: IMO the call site could be improved by changing this

okay: `[self trailerNetworkRequest];`
good: `[self fetchTrailer];`
better: 
```
self.trailerRequest = [self createFetchTrailerRequest]; // the benefit is that you can also cancel the request if it's stored
[self.trailerRequest resume];
```
best:
```
// offload the work to a different object with a single responsibility
// this will help when you have more complex behavior like custom caching, whatever.
self.movieTrailer = [self.movieNetworker fetchTrailer]; 
// then can easily control the request
[self.movieNetwork cancelTrailerRequest];
```

File: Flixter/TrailerViewController.m [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    // Find the trailer inside array
+    int trailerIndex = -1, index = 0;
+    while (index < self.movieTrailerData.count && trailerIndex == -1) {
+        if ([self.movieTrailerData[index][@"type"] isEqual: @"Trailer"]) {
+            trailerIndex = index;
+        }
+        index++;
+    }
+    
+    if (trailerIndex == -1) {
```
Comment (joesus) [inline Flixter/TrailerViewController.m:54]: Doesn't this mean it didn't exist? Why not exit then?

**Review by joesus — COMMENTED** [review]
This is great! Still a lot of trusting the server though. Let's find ways to break this habit. :D
